src/diffusers/loaders/single_file.py

```diff
@@ -12,143 +12,126 @@
         )
 
-        return scheduler_components
+    elif is_transformers_model and is_clip_model_in_single_file(class_obj, checkpoint):
+        load_method = create_diffusers_clip_model_from_ldm
+        config = class_obj.config_class.from_pretrained(pretrained_model_name_or_path, subfolder=name, **kwargs)
 
-    if component_name in ["text_encoder", "text_encoder_2", "tokenizer", "tokenizer_2"]:
-        text_encoder_components = create_text_encoders_and_tokenizers_from_ldm(
-            original_config,
-            checkpoint,
-            model_type=model_type,
+        loaded_sub_model = load_method(
+            class_obj,
+            checkpoint=checkpoint,
+            subfolder=name,
+            config=config,
+            torch_dtype=torch_dtype,
+            local_files_only=local_files_only,
+            **kwargs,
+        )
+
+    else:
+        load_method = getattr(class_obj, "from_pretrained")
+        loaded_sub_model = load_method(
+            pretrained_model_name_or_path,
+            subfolder=name,
             local_files_only=local_files_only,
             torch_dtype=torch_dtype,
         )
-        return text_encoder_components
 
-    if component_name == "safety_checker":
-        if load_safety_checker:
-            from ..pipelines.stable_diffusion.safety_checker import StableDiffusionSafetyChecker
+    return loaded_sub_model
 
-            safety_checker = StableDiffusionSafetyChecker.from_pretrained(
-                "CompVis/stable-diffusion-safety-checker", local_files_only=local_files_only, torch_dtype=torch_dtype
-            )
-        else:
-            safety_checker = None
-        return {"safety_checker": safety_checker}
 
-    if component_name == "feature_extractor":
-        if load_safety_checker:
-            feature_extractor = AutoFeatureExtractor.from_pretrained(
-                "CompVis/stable-diffusion-safety-checker", local_files_only=local_files_only
-            )
-        else:
-            feature_extractor = None
-        return {"feature_extractor": feature_extractor}
-
-    return
+def _legacy_load_safety_checker(local_files_only, torch_dtype):
+    # Support for loading safety checker components using the deprecated
+    # `load_safety_checker` argument.
 
+    from ..pipelines.stable_diffusion.safety_checker import StableDiffusionSafetyChecker
 
-def set_additional_components(
-    pipeline_class_name,
-    original_config,
-    checkpoint=None,
-    model_type=None,
-):
-    components = {}
-    if pipeline_class_name in REFINER_PIPELINES:
-        model_type = infer_model_type(original_config, checkpoint=checkpoint, model_type=model_type)
-        is_refiner = model_type == "SDXL-Refiner"
-        components.update(
-            {
-                "requires_aesthetics_score": is_refiner,
-                "force_zeros_for_empty_prompt": False if is_refiner else True,
-            }
-        )
+    feature_extractor = AutoFeatureExtractor.from_pretrained(
```

Comment:
You should use `AutoImageProcessor` instead since `AutoFeatureExtractor` is deprecated.